src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -871,46 +877,92 @@ void makeNotNullMembersMaybeNull()
                     }
                 }
 
+                return;
+
                 ImmutableArray<Symbol> getMembersNeedingDefaultInitialState()
                 {
                     if (_hasInitialState)
                     {
                         return ImmutableArray<Symbol>.Empty;
                     }
 
+                    bool includeRequiredMembers = true;
+                    bool includeBaseRequiredMembers = true;
+                    bool hasThisConstructorInitializer = false;
+
+                    if (method is SourceMemberMethodSymbol { SyntaxNode: ConstructorDeclarationSyntax { Initializer: { RawKind: var initializerKind } } })
+                    {
+                        var baseOrThisInitializer = (_baseOrThisInitializer ?? GetConstructorThisOrBaseSymbol(this.methodMainNode));
```

Comment:
I didn't follow, why do we need two ways of getting the base/this initializer symbol?

I probably could use a walkthrough when you're back. A few background questions:
1. in current codebase, why do we analyze initializers separately with this "initial state" gymnastics, if you know? (the obvious alternative seems to stick the initializers at the beginning of the body and analyze the whole thing in one pass)
2. given the current two parts design, where does the `SetsRequiredMembers` of `:this()` or `:base()` come into play? Is it in the first or second part?

Reply:
1. Looks like you covered that with Rikki.
2. Always the first analysis pass.